sdv/single_table/base.py

```diff
@@ -429,6 +433,7 @@ def auto_assign_transformers(self, data):
                 The raw data (before any transformations) that will be used to fit the model.
         """
         self.validate(data)
+        data = self._transform_helper(data)
```

Comment:
This looks good. The only thing I'm wondering is if we can somehow streamline these two functions because the validate method fits and transforms the data with the constraints already, so we shouldn't need to do it again in the transform helper